### DISCUSSION TITLE
Fork process worker found in dead state when attempting to run login lookup plugin

### DISCUSSION THREAD

**Discussion by artis3n** [ansible-collections/community.hashi_vault#380] [Q&A] [answered]
When trying to use `community.hashi_vault.vault_login`, I am receiving the following error:

```
TASK [Vault | Login] ******************************************************************************************************
objc[35867]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called.
objc[35867]: +[__NSCFConstantString initialize] may have been in progress in another thread when fork() was called. We cannot safely call it or ignore it in the fork() child process. Crashing instead. Set a breakpoint on objc_initializeAfterForkError to debug.
ERROR! A worker was found in a dead state
```

This error is consistent between `community.hashi_vault.vault_login` and `community.hashi_vault.vault_token_create`.

`ansible-galaxy collection list` tells me I am using `community.hashi_vault 5.0.0`. I also have `hvac 1.1.0` in my poetry's pyproject.toml. I have tried reverting to `hvac 0.11.2` but still get the error.

My task is:

```yaml
- name: Vault | Login
  ansible.builtin.set_fact:
    hashivault_login_data: "{{ lookup('community.hashi_vault.vault_login', url='http://vault:8200', auth_method='approle', role_id='{{ vault_role_id }}', secret_id='{{ vault_secret_id }}') }}"
```

Where `http://vault:8200` is a server in an intranet. I've confirmed I can `vault write auth/approle/login ...` with the role ID and secret ID I'm attempting to use and login against this vault server works fine with the CLI call.

The role ID and secret ID are injected with `-e` during the ansible run, and I've verified they contain the correct contents with `ansible.builtin.debug` calls.

```bash
ansible-playbook -i inventory -e vault_role_id=... -e vault_secret_id=...
```

I have no idea why I'm getting this error. The host I'm running the playbook on is OSX M1.

**Comment by briantist** [comment, marked as answer]
Thank you for including the OS; my first question was going to be if you're on a mac since forking errors with ansible tend to be common there. I don't _think_ this error is actually related to the collection.

You can try setting `--forks=1` just to see if the issue still persists or not.

Here's two old issues that talk about it a bit (though one of them it more pertinent to connecting to Windows hosts, and your issue is in a `set_fact` so not even a module):
- https://github.com/ansible/ansible/issues/32554
- https://github.com/ansible/ansible/issues/32554

If the issue persists, I would recommend raising it in real-time chat over Matrix or IRC:
- https://docs.ansible.com/ansible/latest/community/communication.html#real-time-chat

The general usage and support channel is probably the right place:
- https://matrix.to/#/#users:ansible.com

I'll leave this open for now, in case there's anything that can be done but based on previous experience I don't think that's possible.

Please post an update if you find anything new, thanks!

**Reply by artis3n** [reply]
Thanks for the quick and comprehensive response! The forks parameter didn't help (for anyone who comes to this in the future), but thanks to those resources I was led to https://www.whatan00b.com/posts/debugging-a-segfault-from-ansible/ and using `no_proxy='*'` fixes the apparently underlying Python and Ansible and OSX issue.

I'm now getting an auth error that doesn't make sense because I can `vault write` a successful auth with the creds, but that is well within my own troubleshooting and doesn't appear that anything is wrong in the plugin.

```
fatal: [snuc]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'community.hashi_vault.vault_login'. Error was a <class 'hvac.exceptions.InvalidRequest'>, original message: invalid role ID, on post http://vault:8200/v1/auth/approle/login. invalid role ID, on post http://vault:8200/v1/auth/approle/login"}
```
